libraries/mbed/targets/cmsis/TARGET_WIZNET/TARGET_W7500x/W7500x_i2c.c

```diff
@@ -1,589 +1,403 @@
-}
-
-int I2C_ReceiveData(I2C_TypeDef* I2Cx, int last)    
-{
-    if(last)
-    {   
-        I2C_AcknowledgeConfig(I2Cx,DISABLE);
-        if( I2C_CheckEvent(I2Cx,I2C_ACKT) == ERROR ) {
-            return -1; 
-        }
+    else if(scl_port_num == 3)
+    {
+        HAL_GPIO_Init(GPIOD, &GPIO_InitDef);
+        HAL_GPIO_SetBits(GPIOD, scl_pin_index);
     }
-    else if( I2C_CheckEvent(I2Cx,I2C_ACKT) == ERROR ) {
-            return -1;    
+    else
+    {
+        printf("SCL pin Port number error\r\n");
+        return 1;
     }
-     
-    return (uint8_t)I2Cx -> RXR;    
-}
-
-
-int I2C_Burst_Read(I2C_TypeDef* I2Cx, uint16_t address, uint8_t *data, int length, int stop)
-{
-    int recv_cnt;
-
-    if( I2C_Start(I2Cx,address,I2C_READ_SA7) == ERROR){
-        return -1;
+    
+    //SDA setting
+    GPIO_InitDef.GPIO_Pin = sda_pin_index;
+    GPIO_InitDef.GPIO_Mode = GPIO_Mode_IN;
+    if(sda_port_num == 0)
+    {
+        HAL_GPIO_Init(GPIOA, &GPIO_InitDef);
+        HAL_GPIO_ResetBits(GPIOA, sda_pin_index);
     }
-
-    for(recv_cnt=0;recv_cnt<length;recv_cnt++)
+    else if(sda_port_num == 1)
     {
+        HAL_GPIO_Init(GPIOB, &GPIO_InitDef);
+        HAL_GPIO_ResetBits(GPIOB, sda_pin_index);
     }
-
-    return recv_cnt;
-}
-
-
-int I2C_Burst_Write(I2C_TypeDef* I2Cx, uint16_t address, uint8_t *data, int length, int stop)
-{
-    int cnt;
-
-    if( I2C_Start(I2Cx,address,I2C_WRITE_SA7) == ERROR)
+    else if(sda_port_num == 2)
     {
-        return -1;
+        HAL_GPIO_Init(GPIOC, &GPIO_InitDef);
+        HAL_GPIO_ResetBits(GPIOC, sda_pin_index);
     }
-
-    for(cnt=0;cnt<length;cnt++)
+    else if(sda_port_num == 3)
     {
-        if( I2C_SendDataAck(I2Cx,data[cnt]) == ERROR )
-        {
-            I2C_Stop(I2Cx);
-            return -1;
-        }
+        HAL_GPIO_Init(GPIOD, &GPIO_InitDef);
+        HAL_GPIO_ResetBits(GPIOD, sda_pin_index);
     }
-
-    // If not repeated start, send stop
-    if(stop)
+    else
     {
-        I2C_Stop(I2Cx);
+        printf("SDA pin Port number error\r\n");
+        return 1;
     }
+    
+    HAL_PAD_AFConfig(scl_port_num, scl_pin_index, PAD_AF1);
+    HAL_PAD_AFConfig(sda_port_num, sda_pin_index, PAD_AF1);
 
-    return length;
+    return 0;
 }
 
-
-/**
-  * @brief  Generates I2Cx communication START condition.
-  * @param  I2Cx: where x can be 0 or 1 to select the I2C peripheral.
-  * @param  NewState: NewState of the I2C START condition generation.
-  *   This parameter can be: ENABLE or DISABLE.
-  * @retval None.
-  */
-void I2C_GenerateSTART(I2C_TypeDef* I2Cx, FunctionalState NewState)
+void I2C_WriteBitSCL(I2C_ConfigStruct* conf, uint8_t data)
 {
-    if(NewState != DISABLE)     
-    {    
-        I2Cx->CMDR = I2C_CMDR_STA;  
-    }
-    else                        
+    uint32_t scl_port_num = I2C_PORT(conf->scl);
+    uint32_t scl_pin_index = I2C_PIN_INDEX(conf->scl);
+
+    if(scl_port_num == 0)
     {
-        I2Cx->CMDR = I2C_CMDR_STA;
+        if(data == 1)
```

Comment:
Is this just a pointer to GPIOA peripheral? We could eliminate this big if else by using an offset based on the scl_port_num?  GPIOA + (scl_port_num * the address difference between gpiox) ?